### DISCUSSION TITLE
Getting more live data

### DISCUSSION THREAD

**Discussion by Jotne** [BeatsuDev/tibber.py#43] [General]
After some reading and testing, I do now get my live data, but only some of them:

This works fine and give time at reading and current power:
` print(data.timestamp , data.power)`

But when I try to get some other data, it does not work.  Like voltagePhase1
` print(data.voltagePhase1)`
or 
` print(data.timestamp , data.power , data.voltagePhase1)`

Error message I do get is:
`Retrying to subscribe with backoff. Running <bound method TibberHome.run_websocket_loop of <tibber.types.home.TibberHome object at 0x7fcc2a197130>> in 0.4 seconds after 1 tries.`

If I do test this with my token over at developer.tibber.com, I do get the value for Voltage Phase 1 and other.
I would like to have  all the Phases, both voltage and current.

What do I do wrong?

**Comment by ericbstie** [comment]
Hi, the error message is very poor - I'll have to look into that. Though perhaps the error is that the properties should be snake_case instead of camelCase? E.g. `data.voltage_phase_1` instead of `data.voltagePhase1`. Let me know if that works!

**Comment by Jotne** [comment]
Thanks, that did the trick.  This line then works fine:
` print(data.voltage_phase_1 , data.voltage_phase_2 , data.voltage_phase_3 , data.power , data.max_power , data.last_meter_consumption)`

Strange that it does not follow the man pages: https://developer.tibber.com/docs/reference

But I am not able to get the current.  None of these do work:
`data.current_phase_1
data.current_phase1
data.currentPhase1
data.current_l_1
data.currentL1
data.current_l1`

Web show that its **currentL1** that should be used.

Also how to I write the output to a file instead of screen?
I do not now much about python and all I tried only gives errors, so tried to redirect, and got it working like this:

`python3 -u live.py > /var/log/tibber.csv`

Its some ugly compare to that python can write by it self.

**Reply by ericbstie** [reply]
data.currentL1 should be correct: https://github.com/BeatsuDev/tibber.py/blob/master/tibber/types/live_measurement.py#L143. Do you get any errors? It's working fine with the demo token:

```py
import tibber

account = tibber.Account(tibber.DEMO_TOKEN)
home = account.homes[0]

@home.event("live_measurement")
async def show_accumulated_cost(data):
  print(f"{data.currentL1}")

home.start_live_feed(user_agent = "UserAgent/0.0.1") 
```

If you want all the output from your script exactly as it is shown in the terminal, but into a file, the solution you came up with should be correct and do exactly that. If you want to format it in some other way, you can check out the [`csv`](https://docs.python.org/3/library/csv.html) library or [`pandas`](https://pandas.pydata.org/docs/getting_started/index.html) and then write directly to a file within the script.

**Comment by Jotne** [comment]
How stupid one can be ;)
Since you added _ in front of all upper case letter and change all uppercase to lower case, I did not test the original value.  Now it works fine.  I would then use the redirect.  Thanks for all the help
